hs_core/models.py

```diff
@@ -86,7 +87,11 @@ def can_add(self, request):
     def can_delete(self, request):
```

Comment:
@hyi You'll need to add test cases for authorize() to cover the new behavior.  Let me know if I can help.